pollux/core/src/main/scala/org/hyperledger/identus/pollux/core/service/uriResolvers/HttpUrlResolver.scala

```diff
@@ -0,0 +1,75 @@
+package org.hyperledger.identus.pollux.core.service.uriResolvers
+
+import org.hyperledger.identus.shared.http.{GenericUriResolverError, UriResolver}
+import org.hyperledger.identus.shared.models.StatusCode
+import zio.*
+import zio.http.*
+
+import java.nio.charset.StandardCharsets
+
+class HttpUrlResolver(client: Client) extends UriResolver {
+  import HttpUriResolver.*
+  override def resolve(uri: String): IO[GenericUriResolverError, String] = {
+    val program = for {
+      url <- ZIO.fromEither(URL.decode(uri)).mapError(_ => InvalidURI(uri))
+      response <- client
+        .request(Request(url = url))
+        .mapError(t => ConnectionError(t.getMessage))
+      body <- response.status match {
+        case Status.Ok =>
+          response.body.asString.mapError(t => ResponseProcessingError(t.getMessage))
+        case Status.NotFound =>
+          ZIO.fail(ResourceNotFound(uri))
+        case status if status.isError =>
+          response.body.asStream
+            .take(1024) // Only take the first 1024 bytes from the response body (if any).
+            .runCollect
+            .map(c => new String(c.toArray, StandardCharsets.UTF_8))
+            .orDie
+            .flatMap(errorMessage => ZIO.fail(UnexpectedUpstreamResponseReceived(status.code, Some(errorMessage))))
+        case status =>
+          ZIO.fail(UnexpectedUpstreamResponseReceived(status.code))
+      }
+    } yield body
+    program.provideSomeLayer(zio.Scope.default)
```

Comment:
What is the purpose of the `Scope.default` here?

Reply:
I'm not sure It is a refactored code, but if I recall, ZIO HTTP client needs this layer to function